src/features/hotspots/setup/ConfirmLocationScreen.tsx

```diff
@@ -0,0 +1,123 @@
+import { useNavigation } from '@react-navigation/native'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import BackScreen from '../../../components/BackScreen'
+import Box from '../../../components/Box'
+import Button from '../../../components/Button'
+import ImageBox from '../../../components/ImageBox'
+import Map from '../../../components/Map'
+import Text from '../../../components/Text'
+import { reverseGeocode } from '../../../utils/location'
+import { HotspotSetupNavigationProp } from './hotspotSetupTypes'
+
+const ConfirmLocationScreen = () => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<HotspotSetupNavigationProp>()
+  const [mapCenter, setMapCenter] = useState([122.419, 37.775])
+  const [markerCenter, setMarkerCenter] = useState([0, 0])
+  const [mapMoving, setMapMoving] = useState(false)
+  const [hasGPSLocation, setHasGPSLocation] = useState(false)
+  const [locationName, setLocationName] = useState('')
+  const [zoomLevel, setZoomLevel] = useState(2)
+
+  const onMapMoved = async (newCoords: number[]) => {
+    setMarkerCenter(newCoords)
+    setMapMoving(false)
+
+    const [longitude, latitude] = newCoords
+    const [{ street, city }] = await reverseGeocode(latitude, longitude)
+    const name = street && city ? [street, city].join(', ') : 'Loading...'
+    setLocationName(name)
+
```

Comment:
@tyler-whitman @allenan ↓↓

Reply:
I think that was to make the next button be disabled for 3 seconds or until the user moves the map. We had a problem where users were just pressing "next" "next" "next" to get through onboarding and ended up asserting in the wrong location

Reply:
oh, ok, that makes sense. I'll add that in.